core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/HealthCheckedEndpointGroup.java

```diff
@@ -75,6 +76,11 @@ public final class HealthCheckedEndpointGroup extends DynamicEndpointGroup {
 
     private static final Logger logger = LoggerFactory.getLogger(HealthCheckedEndpointGroup.class);
 
+    /**
+     * Lock for {@link contextGroupChain}.
+     */
+    private final ReentrantLock lock = new ReentrantLock();
```

Comment:
Should we move this below static methods? https://armeria.dev/community/developer-guide#organize